### PR TITLE
samples: net: mqtt_publisher: Add section on connecting with TLS

### PR DESCRIPTION
**Pull request by vanti**
Adding a section to the sample's documentation to address what I had in mind in the discussion in PR #12715. It gives some specific instructions on how to try it with TLS enabled. This should be helpful
for users who are not knowledgeable with regards to creating/setting up TLS certificates locally.

### CODE REVIEW

File: samples/net/mqtt_publisher/README.rst [hunk 1]
```diff
@@ -114,6 +114,32 @@ Open another terminal window and type:
 
 	$ mosquitto_sub -t sensors
 
+Connecting securely using TLS
+=============================
+
+While it is possible to set up a local secure MQTT server and update the
+sample to connect to it, it does require some work on the user's part to
+create the certificates and to set them up with the server. The user is
```
Comment (dbkinder) [inline samples/net/mqtt_publisher/README.rst]: I'd delete this last sentence, "The user is welcome to do it..."

File: samples/net/mqtt_publisher/README.rst [hunk 2]
```diff
@@ -114,6 +114,32 @@ Open another terminal window and type:
...
+welcome to do it if they are knowledgeable in this area.
+
+Alternatively, a quicker way to try this sample with TLS enabled may be to
```
Comment (dbkinder) [inline samples/net/mqtt_publisher/README.rst]: Let's give some more links to information about this service:
```
Alternatively a `publicly available Mosquitto MQTT server/broker
<https://test.mosquitto.org/>  is available to quickly and easily
try this sample with TLS enabled, by following these steps:
```

File: samples/net/mqtt_publisher/README.rst [hunk 3]
```diff
@@ -114,6 +114,32 @@ Open another terminal window and type:
...
+  contents (or set it to its filename if the socket offloading feature is
+  enabled on your platform and :option:`CONFIG_TLS_CREDENTIAL_FILENAMES` is
+  set to ``y``).
+- In :file:`src/config.h`, set SERVER_ADDR to the IP address to connect to,
+  ie. the IP address of test.mosquitto.org ``"37.187.106.16"``
+- In :file:`src/main.c`, set TLS_SNI_HOSTNAME to ``"test.mosquitto.org"``
+  to match the Common Name (CN) in the downloaded certificate.
+- Build the sample by specifying ``-DOVERLAY_CONFIG=overlay-tls.conf``
+  when running cmake (or refer to the TLS offloading section below if
+  your platform uses the offloading feature instead).
```
Comment (dbkinder) [inline samples/net/mqtt_publisher/README.rst]:
```suggestion
  your platform uses the offloading feature).
```

File: samples/net/mqtt_publisher/README.rst [hunk 4]
```diff
@@ -114,6 +114,32 @@ Open another terminal window and type:
...
+- Run the sample.
+
```
Comment (dbkinder) [inline samples/net/mqtt_publisher/README.rst]: add the command line to do this...  (is it ``$ mosquitto_sub -t sensors``)?

**Review by dbkinder — CHANGES_REQUESTED** [review]

**Review by rlubos — APPROVED** [review]
Thanks.

File: samples/net/mqtt_publisher/README.rst (hunk 4 above)
Comment (vanti) [reply samples/net/mqtt_publisher/README.rst]: I meant running the mqtt_publisher sample itself: `ninja flash`

**Comment by vanti** [comment]
Thanks for the review. Updated based on comments from @dbkinder.

File: samples/net/mqtt_publisher/README.rst (hunk 4 above)
Comment (nashif) [reply samples/net/mqtt_publisher/README.rst]: Then you probably want to change this to "Flash the binary into the device" instead of "run the sample"

Comment (vanti) [reply samples/net/mqtt_publisher/README.rst]: Ok, changed to "Flash the binary onto the device to run the sample"

**Review by jukkar — APPROVED** [review]
LGTM

**Comment by vanti** [comment]
Updated per comment from @nashif, thanks.

**Review by nashif — APPROVED** [review]

**Pull request merged**